lib/kanta/query.ex

```diff
@@ -293,6 +294,31 @@ defmodule Kanta.Query do
       def search_query(query, ""), do: query
 
       def search_query(query, search) do
+        repo = Repo.get_repo()
+
+        if Postgresql.migrated_version(%{repo: repo}) >= 2 do
+          search_query_fuzzy(query, search)
+        else
+          search_query_legacy(query, search)
+        end
+      end
+
+      defp search_query_fuzzy(query, search) do
+        from(s in unquote(opts[:module]),
+          where:
+            fragment(
+              "msgid ILIKE ? OR (SOUNDEX(msgid) = SOUNDEX(?) AND LEVENSHTEIN(?, msgid) <= 3)",
+              ^("%" <> search <> "%"),
+              ^search,
+              ^search
```

Comment:
I don't see `searchable` mentioned here?
You should form the query `where` using `to_tsquery()` and use `@@` operator for matching against the `searchable`.
Then order the results using `ts_rank(...)`

Example:

```elixir
  defmacro form_search_query(search_term) do
    quote do
      fragment(
        "SELECT to_tsquery(string_agg(unaccent(lexeme) || ':*', ' & ' order by positions)) FROM unnest(to_tsvector(?))",
        unquote(search_term)
      )
    end
  end

  defmacro ts_rank(left, right) do
    quote do
      fragment("ts_rank(?, ?)", unquote(left), unquote(right))
    end
  end

# then somewhere
query
|> 
where(
          [thing_that_holds_searchable: thing],
          fragment("? @@ ?", thing.searchable, form_search_query(^search_term))
        )
|> order_by(
          [thing_that_holds_searchable: thing],
          desc:
            ts_rank(
              thing.searchable,
              form_search_query(^search_term)
            )
        )
```